api/anubis/views/admin/courses.py

```diff
@@ -512,3 +498,57 @@ def admin_course_remove_professor_id(user_id: str):
             "variant": "warning",
         }
     )
+
+
+@courses_.route("/batch/students", methods=["POST"])
+@require_admin()
+@json_endpoint(required_fields=[("students", list), ("create_pvc", bool)])
+def admin_course_batch_students(students: list[dict], create_pvc: bool):
+    """
+    Batch add students to a course.
+
+    :param students: A list of students to add to the course
+    :param create_pvc: Whether or not to create PVCs for the students
+
+    :return:
+    """
+    # Get all students already in the course
+    students_in_course: list[User] = (
+        User.query.join(InCourse)
+        .filter(
+            InCourse.course_id == course_context.id,
+        )
+        .all()
+    )
+
+    for student in students:
+        # Get the user object for the student
+        user: User= User.query.filter(User.netid == student["netid"]).first()
+
+
+        # If the user doesnt exist we create one
+        if not user:
+            user = User(
+                netid=student["netid"],
+                name=student.get("name", "John Doe"),
```

Comment:
Can you iterate through the whole student list, and verify the structure before. Reject if anything is missing or malformed.

Reply:
Sure I am validating the json on the fe right now, maybe I should just check field there.

Reply:
You can use [anubis.utils.data.verify_data_shape](https://github.com/AnubisLMS/Anubis/blob/8e911416997cd8ca7ec100bdfab0563bb1694081/api/anubis/utils/data.py#L45) with [anubis.utils.data.req_assert](https://github.com/AnubisLMS/Anubis/blob/8e911416997cd8ca7ec100bdfab0563bb1694081/api/anubis/utils/data.py#L316) for this.